protocol/bc/txissuance.go

```diff
@@ -53,3 +58,58 @@ func NewIssuance(anchor Entry, value AssetAmount, data Hash, ordinal int) *Issua
 	iss.ordinal = ordinal
 	return iss
 }
+
+func (iss *Issuance) checkValid(vs *validationState) error {
+	if iss.Witness.AssetDefinition.InitialBlockID != vs.blockchainID {
+		return errors.WithDetailf(errWrongBlockchain, "current blockchain %x, asset defined on blockchain %x", vs.blockchainID[:], iss.Witness.AssetDefinition.InitialBlockID[:])
+	}
+
+	computedAssetID := iss.Witness.AssetDefinition.ComputeAssetID()
+	if computedAssetID != iss.Body.Value.AssetID {
+		return errors.WithDetailf(errMismatchedAssetID, "asset ID is %x, issuance wants %x", computedAssetID[:], iss.Body.Value.AssetID[:])
+	}
+
+	err := vm.Verify(NewTxVMContext(vs.tx, iss, iss.Witness.AssetDefinition.IssuanceProgram, iss.Witness.Arguments))
+	if err != nil {
+		return errors.Wrap(err, "checking issuance program")
+	}
+
+	var anchored Hash
+	switch a := iss.Anchor.(type) {
```

Comment:
I like this suggestion from a Go perspective, but on the other hand the spec is explicit about the set of types allowed as anchors and I can't decide whether the Go-cleanness of your suggestion is worth hiding the explicitness in the spec.